client/security/src/main/java/org/eclipse/kapua/client/security/amqpclient/Client.java

```diff
@@ -73,12 +77,61 @@ public TextMessage createTextMessage() throws JMSException {
         return session.createTextMessage();
     }
 
-    public boolean isConnected() {
+    public void stop() throws JMSException {
+        active = false;
+        disconnect();
+    }
+
+    private void disconnect() throws JMSException {
+        if (connection != null) {
+            connection.close();
+        }
+    }
+
+    private void connect() {
+        int connectAttempt = 0;
+        while (active && !isConnected()) {
+            logger.info("Connect attempt {}...", connectAttempt);
+            try {
+                logger.info("Service client {} - restarting attempt... {}", this, connectAttempt);
+                disconnect();
+                connection = connectionFactory.createConnection();
+                connection.setExceptionListener(exceptionListener);
+                connection.setClientID(clientId);
+                connection.start();
+                session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+                logger.info("AMQP client binding request sender to: {}", requestAddress);
+                logger.info("AMQP client binding message listener to: {}", replyAddress);
+                consumer = session.createConsumer(session.createQueue(replyAddress));
+                consumer.setMessageListener(clientMessageListener);
+                producer = session.createProducer(session.createQueue(requestAddress));
+                clientMessageListener.init(session, producer);
+                connectionStatus = true;
+                logger.info("Service client {} - restarting attempt... {} DONE (Connection restored)", this, connectAttempt);
+            } catch (JMSException e) {
+                logger.info("Connect attempt {}... FAIL", connectAttempt, e);
+                //wait a bit
+                waitBeforeRetry();
+            }
+            connectAttempt++;
+        }
+    }
+
+    private boolean isConnected() {
         try {
-            return connection!=null && connection.getClientID()!=null && connectionStatus.isAlive();
+            return connection!=null && connection.getClientID()!=null && connectionStatus;
         } catch (JMSException e) {
             logger.warn("Error while validating connection: {}", e.getMessage(), e);
             return false;
         }
     }
+
+    private void waitBeforeRetry() {
+        try {
+            Thread.sleep(WAIT_BETWEEN_RECONNECTION_ATTEMPT);
```

Comment:
Should we add a backoff function here, instead of a fixed sleep?

Reply:
With a backoff we could define a maximum value, but I'm okay proceeding with this simpler solution unless we find a need to change it otherwise.